app/src/main/java/com/imashnake/animite/features/searchbar/SearchFrontDrop.kt

```diff
@@ -0,0 +1,312 @@
+import androidx.compose.foundation.layout.consumeWindowInsets
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.imeNestedScroll
+import androidx.compose.foundation.layout.imePadding
+import androidx.compose.foundation.layout.navigationBarsPadding
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.size
+import androidx.compose.foundation.layout.statusBarsPadding
+import androidx.compose.foundation.layout.width
+import androidx.compose.foundation.lazy.LazyColumn
+import androidx.compose.foundation.shape.CircleShape
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.foundation.text.KeyboardOptions
+import androidx.compose.material.ContentAlpha
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.rounded.Close
+import androidx.compose.material.icons.rounded.KeyboardArrowRight
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.Surface
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.draw.drawBehind
+import androidx.compose.ui.focus.FocusRequester
+import androidx.compose.ui.focus.focusRequester
+import androidx.compose.ui.graphics.vector.ImageVector
+import androidx.compose.ui.res.dimensionResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.res.vectorResource
+import androidx.compose.ui.text.input.ImeAction
+import androidx.compose.ui.text.style.TextOverflow
+import androidx.compose.ui.unit.Dp
+import androidx.compose.ui.unit.dp
+import androidx.lifecycle.viewmodel.compose.viewModel
+import com.imashnake.animite.R
+import com.imashnake.animite.SearchQuery
+import com.imashnake.animite.core.ui.Icon
+import com.imashnake.animite.core.ui.IconButton
+import com.imashnake.animite.core.ui.TextField
+import com.imashnake.animite.dev.internal.Constants
+import com.imashnake.animite.features.ui.MediaSmall
+import com.imashnake.animite.type.MediaFormat
+
+/**
+ * TODO: Kdoc
+ * @param hasExtraPadding TODO: This causes recomposition! Probably wrap it in a lambda.
+ */
+@OptIn(ExperimentalAnimationApi::class, ExperimentalLayoutApi::class)
+@Composable
+fun SearchFrontDrop(
+    modifier: Modifier = Modifier,
+    viewModel: SearchViewModel = viewModel(),
+    hasExtraPadding: Boolean,
+    onItemClick: (Int?) -> Unit
+) {
+    var isExpanded by remember { mutableStateOf(false) }
+    // This should ideally be in a modifier lambda to prevent recomposition.
```

Comment:
Animations are supposed to cause recomposition, otherwise things wouldn't animate

Reply:
o7